README.md

```diff
@@ -46,13 +48,21 @@ For more information on Puppet module installation see [Puppet Labs: Installing
 
 PuppetLabs provides NetDev resource support for Cisco Nexus devices with their [`puppetlabs-netdev-stdlib`](https://forge.puppet.com/puppetlabs/netdev_stdlib) module. Installing the `ciscopuppet` module automatically installs both the `ciscopuppet` and `netdev_stdlib` modules.
 
-### <a name="setup-puppet-agent">Puppet Agent<a>
+##### The `puppetlabs-resource_api` module
+
+The 'ciscopuppet' module has a dependency on the `resource_api`. The Resource API is automatically installed when you install the module. Alternatively, it can be manually installed by running `puppet module install puppetlabs-resource_api`, or by following the setup instructions in the [Resource API README](https://github.com/puppetlabs/puppetlabs-resource_api#resource_api).
+
+### <a name="setup-puppet-agent">Puppet Agent (LTS)</a>
 
 The Puppet Agent requires installation and setup on each device. Agent setup can be performed as a manual process or it may be automated. For more information please see the [README-agent-install.md][USER-1] document for detailed instructions on agent installation and configuration on Cisco Nexus devices.
 
 ##### The `cisco_node_utils` Ruby Gem
 
-The [`cisco_node_utils`](https://rubygems.org/gems/cisco_node_utils) ruby gem is a required component of the `ciscopuppet` module. This gem contains platform APIs for interfacing between Cisco CLI and Puppet agent resources. The gem can be automatically installed by Puppet agent by simply using the [`ciscopuppet::install`](https://github.com/cisco/cisco-network-puppet-module/blob/master/examples/demo_all_cisco.pp#L19) helper class, or it can be installed manually.
+The [`cisco_node_utils`](https://rubygems.org/gems/cisco_node_utils) Ruby gem is a required component of the `ciscopuppet` module. This gem contains platform APIs for interfacing between Cisco CLI and Puppet agent resources. The gem can be automatically installed by Puppet agent by simply using the [`ciscopuppet::install`](https://github.com/cisco/cisco-network-puppet-module/blob/master/examples/demo_all_cisco.pp#L19) helper class, or it can be installed manually.
+
+##### The `puppet-resource_api` Ruby Gem
+
+The [`puppet-resource_api`](https://rubygems.org/gems/puppet-resource_api) Ruby gem is a required component of the `ciscopuppet` module. Puppet automatically installs the gem using the [`ciscopuppet::install`](https://github.com/cisco/cisco-network-puppet-module/blob/master/examples/demo_all_cisco.pp#L19) helper class, or it can be installed manually.

```

Comment:
This is repeating the "`ciscopuppet::install`" class thing , and is linking to (bad) example code. Meanwhile we made the `install` class a first-class member of the module (https://github.com/puppetlabs/cisco-network-puppet-module/blob/b0e9159ecc13c958a442e361b9afdbdf935af865/manifests/install.pp) and we should be referencing its documentation directly.

Please restructure this paragraph and the one above to lead in with "Use ciscopuppet::install to ..." and end with " .. only in exceptional circumstances you should consider installing the dependencies manually."